src/Symfony/Component/Asset/Context/NullContext.php

```diff
@@ -18,6 +18,13 @@
  */
 class NullContext implements ContextInterface
 {
+    public function __construct()
+    {
+        if (__CLASS__ !== get_class($this)) {
```

Comment:
>without talking about the fact that having an annotation for a keyword supported by php looks weird to me :P

I strongly disagree here: if we were in a perfect world no problem, but if you are using proxy manager for example you can't proxy a final service because the proxying is done via inheritance. There is also the case where for practical reason you can't declare it as final, but you wish to convey the information that a given class/method is not meant to be extended/overridden.

And for the IDE auto-completion, Symfony used to use the `@public` tag for a while and `@internal` or `@private` as equally used. IDE support comes with usage, not vice-versa.